### ISSUE TITLE
idafree uninstall Bad CPU type error

### ISSUE THREAD

**Issue by mshavliuk** [Homebrew/homebrew-cask#70983]
#### General troubleshooting steps

- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [x] I have retried my command with `--force`.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

#### Description of issue

[Idafree](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/idafree.rb) uninstall fails.

#### Command that failed

```
brew uninstall idafree
```

#### Output of command with `--force --verbose --debug`

```
==> Uninstalling Cask idafree
==> Uninstalling Cask idafree
==> Uninstalling artifacts
==> 2 artifact/s defined
#<SortedSet: {#<Cask::Artifact::Uninstall:0x00007fae7238de98 @cask=#<Cask::Cask:0x00007fae72376388 @token="idafree", @sourcefile_path=#<Pathname:/usr/local/Caskroom/idafree/.metadata/7.0/20190309211946.370/Casks/idafree.rb>, @tap=nil, @block=#<Proc:0x00007fae72376298@/usr/local/Caskroom/idafree/.metadata/7.0/20190309211946.370/Casks/idafree.rb:1>, @caskroom_path=#<Pathname:/usr/local/Caskroom/idafree>, @metadata_master_container_path=#<Pathname:/usr/local/Caskroom/idafree/.metadata>, @config=#<Cask::Config:0x00007fae7237f258 @default={:appdir=>#<Pathname:/Applications>, :prefpanedir=>#<Pathname:/Users/mshavliuk/Library/PreferencePanes>, :qlplugindir=>#<Pathname:/Users/mshavliuk/Library/QuickLook>, :dictionarydir=>#<Pathname:/Users/mshavliuk/Library/Dictionaries>, :fontdir=>#<Pathname:/Users/mshavliuk/Library/Fonts>, :colorpickerdir=>#<Pathname:/Users/mshavliuk/Library/ColorPickers>, :servicedir=>#<Pathname:/Users/mshavliuk/Library/Services>, :input_methoddir=>#<Pathname:/Users/mshavliuk/Library/Input Methods>, :internet_plugindir=>#<Pathname:/Users/mshavliuk/Library/Internet Plug-Ins>, :audio_unit_plugindir=>#<Pathname:/Users/mshavliuk/Library/Audio/Plug-Ins/Components>, :vst_plugindir=>#<Pathname:/Users/mshavliuk/Library/Audio/Plug-Ins/VST>, :vst3_plugindir=>#<Pathname:/Users/mshavliuk/Library/Audio/Plug-Ins/VST3>, :screen_saverdir=>#<Pathname:/Users/mshavliuk/Library/Screen Savers>}, @env={}, @explicit={}>, @dsl=#<Cask::DSL:0x00007fae7237c2d8 @cask=#<Cask::Cask:0x00007fae72376388 ...>, @token="idafree", @version="7.0", @sha256="e1101908ccb13eab5b33d7b4cb8c991993a7054a23232f1c94e214e228569ba8", @url=#<URL:0x00007fae72386e68 @uri=#<URI::HTTPS https://out7.hex-rays.com/files/idafree70_mac.tgz>, @user_agent=:default, @specs={}>, @name=["IDA Free"], @homepage="https://www.hex-rays.com/index.shtml", @artifacts=#<SortedSet: {...}>, @language=nil>>, @directives={:script=>{:executable=>"/Applications/IDA Free/uninstall.app/Contents/MacOS/installbuilder.sh", :args=>["--mode", "unattended"], :sudo=>true}, :signal=>[]}>, #<Cask::Artifact::Installer:0x00007fae723842a8 @cask=#<Cask::Cask:0x00007fae72376388 @token="idafree", @sourcefile_path=#<Pathname:/usr/local/Caskroom/idafree/.metadata/7.0/20190309211946.370/Casks/idafree.rb>, @tap=nil, @block=#<Proc:0x00007fae72376298@/usr/local/Caskroom/idafree/.metadata/7.0/20190309211946.370/Casks/idafree.rb:1>, @caskroom_path=#<Pathname:/usr/local/Caskroom/idafree>, @metadata_master_container_path=#<Pathname:/usr/local/Caskroom/idafree/.metadata>, @config=#<Cask::Config:0x00007fae7237f258 @default={:appdir=>#<Pathname:/Applications>, :prefpanedir=>#<Pathname:/Users/mshavliuk/Library/PreferencePanes>, :qlplugindir=>#<Pathname:/Users/mshavliuk/Library/QuickLook>, :dictionarydir=>#<Pathname:/Users/mshavliuk/Library/Dictionaries>, :fontdir=>#<Pathname:/Users/mshavliuk/Library/Fonts>, :colorpickerdir=>#<Pathname:/Users/mshavliuk/Library/ColorPickers>, :servicedir=>#<Pathname:/Users/mshavliuk/Library/Services>, :input_methoddir=>#<Pathname:/Users/mshavliuk/Library/Input Methods>, :internet_plugindir=>#<Pathname:/Users/mshavliuk/Library/Internet Plug-Ins>, :audio_unit_plugindir=>#<Pathname:/Users/mshavliuk/Library/Audio/Plug-Ins/Components>, :vst_plugindir=>#<Pathname:/Users/mshavliuk/Library/Audio/Plug-Ins/VST>, :vst3_plugindir=>#<Pathname:/Users/mshavliuk/Library/Audio/Plug-Ins/VST3>, :screen_saverdir=>#<Pathname:/Users/mshavliuk/Library/Screen Savers>}, @env={}, @explicit={}>, @dsl=#<Cask::DSL:0x00007fae7237c2d8 @cask=#<Cask::Cask:0x00007fae72376388 ...>, @token="idafree", @version="7.0", @sha256="e1101908ccb13eab5b33d7b4cb8c991993a7054a23232f1c94e214e228569ba8", @url=#<URL:0x00007fae72386e68 @uri=#<URI::HTTPS https://out7.hex-rays.com/files/idafree70_mac.tgz>, @user_agent=:default, @specs={}>, @name=["IDA Free"], @homepage="https://www.hex-rays.com/index.shtml", @artifacts=#<SortedSet: {...}>, @language=nil>>, @args={:must_succeed=>true, :sudo=>false, :args=>["--mode", "unattended"], :print_stdout=>true}, @path=#<Pathname:idafree-7.0-osx-installer.app/Contents/MacOS/installbuilder.sh>>}>
==> Uninstalling artifact of class Cask::Artifact::Uninstall
==> Running uninstall script /Applications/IDA Free/uninstall.app/Contents/MacOS/installbuilder.sh
/usr/bin/sudo -E -- /Applications/IDA\ Free/uninstall.app/Contents/MacOS/installbuilder.sh --mode unattended
/Applications/IDA Free/uninstall.app/Contents/MacOS/installbuilder.sh: line 20: /Applications/IDA Free/uninstall.app/Contents/MacOS/osx-intel: Bad CPU type in executable
Error: Failure while executing; `/usr/bin/sudo -E -- /Applications/IDA\ Free/uninstall.app/Contents/MacOS/installbuilder.sh --mode unattended` exited with 126. Here's the output:
/Applications/IDA Free/uninstall.app/Contents/MacOS/installbuilder.sh: line 20: /Applications/IDA Free/uninstall.app/Contents/MacOS/osx-intel: Bad CPU type in executable

Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/usr/local/Homebrew/Library/Homebrew/system_command.rb:176:in `assert_success!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:53:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:29:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:301:in `uninstall_script'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/abstract_uninstall.rb:70:in `dispatch_uninstall_directive'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:11:in `block in uninstall_phase'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:10:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/uninstall.rb:10:in `uninstall_phase'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:429:in `block in uninstall_artifacts'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:777:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:777:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:425:in `uninstall_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:376:in `uninstall'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:24:in `block in run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:14:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/uninstall.rb:14:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/abstract_command.rb:36:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:92:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:158:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:123:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:9:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:102:in `<main>'
Error: Kernel.exit
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:167:in `exit'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:167:in `rescue in run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:146:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:123:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:9:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:102:in `<main>'
```

#### Output of `brew cask doctor`

```
==> Homebrew Version
2.1.14-44-g2bf8015
==> macOS
10.15
==> SIP
Enabled
==> Java
N/A
==> xattr issues
<NONE>
==> Gatekeeper support
Error: Unknown support status
==> Homebrew Cask Install Location
<NONE>
==> Homebrew Cask Staging Location
/usr/local/Caskroom
==> Homebrew Cask Taps:
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask (3479 casks)
==> Contents of $LOAD_PATH
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-macho-2.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-rspec-1.36.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-performance-1.5.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rubocop-0.75.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unicode-display_width-1.6.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ruby-progressbar-1.10.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-wait-0.0.9/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-retry-0.6.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-its-1.3.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-3.9.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-mocks-3.9.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-expectations-3.9.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-core-3.9.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rspec-support-3.9.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ronn-0.7.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rdiscount-2.2.0.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-13/2.6.0-static/rdiscount-2.2.0.1
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/rainbow-3.0.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/plist-3.5.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parser-2.6.5.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parallel_tests-2.29.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/parallel-1.18.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mustache-1.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mechanize-2.7.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/webrobots-0.1.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ntlm-http-0.1.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/nokogiri-1.10.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-13/2.6.0-static/nokogiri-1.10.4
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mini_portile2-2.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-persistent-3.1.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/net-http-digest_auth-1.4.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-3.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/mime-types-data-3.2019.1009/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/jaro_winkler-1.5.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-13/2.6.0-static/jaro_winkler-1.5.3
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/http-cookie-1.0.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/hpricot-0.8.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-13/2.6.0-static/hpricot-0.8.6
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/domain_name-0.5.20190701/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf-0.1.4/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/unf_ext-0.0.7.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-13/2.6.0-static/unf_ext-0.0.7.6
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/diff-lcs-1.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/coveralls-0.8.23/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/thor-0.20.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/term-ansicolor-1.7.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tins-1.21.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-0.16.1/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/simplecov-html-0.10.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/docile-1.3.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/json-2.2.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/extensions/x86_64-darwin-13/2.6.0-static/json-2.2.0
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/connection_pool-2.2.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/ast-2.4.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/activesupport-5.2.3/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/tzinfo-1.2.5/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/thread_safe-0.3.6/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/minitest-5.12.2/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/i18n-1.7.0/lib
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/bundler/../ruby/2.6.0/gems/concurrent-ruby-1.1.5/lib
/Library/Ruby/Site/2.6.0
/Library/Ruby/Site/2.6.0/x86_64-darwin19
/Library/Ruby/Site/2.6.0/universal-darwin19
/Library/Ruby/Site
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0/x86_64-darwin19
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby/2.6.0/universal-darwin19
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/vendor_ruby
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/x86_64-darwin19
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/universal-darwin19
/usr/local/Homebrew/Library/Homebrew
==> Environment Variables
LC_ALL="en_US.UTF-8"
PATH="/usr/local/Homebrew/Library/Homebrew/shims/scm:/usr/bin:/bin:/usr/sbin:/sbin"
SHELL="/bin/bash"
```

#### Output of `brew tap`

```
heroku/brew
homebrew/cask
homebrew/core
homebrew/services
schniz/tap
tophat/bar
```

**Comment by mshavliuk** [comment]
Oh, I got it: it seems like this issue is caused by [end of support of 32 bit apps by macOS ](https://www.theverge.com/2019/10/12/20908567/apple-macos-catalina-breaking-apps-32-bit-support-how-to-prepare-avoid-update)

**Comment by vitorgalvao** [comment]
Unfortunately, that’s not something we can help with. It’s their script which is failing. Remove the directory at `$(brew --prefix)/Caskroom/idafree` and HBC will forget about it, but to complete the uninstall, you’ll need to contact them. We’d be interested in their answer as well.

**Comment by vitorgalvao** [comment]
Try manually opening `/Applications/IDA Freeware 7.0/uninstall.app` and see if that works.

**Comment by mshavliuk** [comment]
> Try manually opening

I have tried, but it doesn't help. Even something like `$path_to_uninstall --help` was throwing the same error.

> We’d be interested in their answer

I will let you know.

**Comment by vitorgalvao** [comment]
What’s the content of `/Applications/IDA Free/uninstall.app/Contents/MacOS/installbuilder.sh`? You can copy it with `pbcopy < '/Applications/IDA Free/uninstall.app/Contents/MacOS/installbuilder.sh'`.

**Comment by mshavliuk** [comment]
Here is the answer from IDA support:
> Catalina removed support for 32-bit apps and probably your copy is the
old one with 32-bit installer. You can download the current version from
our website but it comes without support so try to resolve any issues on
your own.

Speaking about `installbuilder.sh` content: I have already deleted all files, related to this app. But there were not too many lines: it defined current CPU architecture with `uname -p` and based on that the appropriate (by design at least) bin file was run. In my case, it was some bin app in the `uninstall.app/intel` dir.

**Comment by kkumlien** [comment]
installbuilder.sh should come from BitRock InstallBuilder, you can try getting the latest script and binary (+ related .dat file) from their latest distribution (as of today): https://installbuilder.bitrock.com/installbuilder-19.10.0-osx-installer.dmg

They mention recent fixes for macOS 10.15: https://installbuilder.bitrock.com/changelog.html

**Comment by kkumlien** [comment]
See a solution here for another product based on the InstallBuilder: https://community.jaspersoft.com/wiki/how-run-bundled-installer-jasperreports-server-macos-1015-catalina
